syntax = "proto3";

message Address {
  // The IP address in textual format.
  string ip_address = 1;

  // The port number.
  int32  port       = 2;
}

message ServerConfig {
  // The address on which the server listens for other
  // incoming server connection.
  Address listen_addr = 1;

  // The address on which the server listens for clients.
  Address client_addr = 2;

  // A path to the persistent log file.
  string  log_file_path = 3;
}

message KVConfiguration {
  // A list of servers.
  repeated ServerConfig servers = 1;
}